From: Hilko Bengen <bengen@debian.org>
Date: Wed, 27 Apr 2011 00:18:42 +0200
Subject: Patched out SSL2 support since it is no longer supported by OpenSSL.

Now that openssl has dropped SSLv2 support we need to patch it out.
Patch picked up from debian patchseries 5.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

---
 mailx.1   |    2 +-
 openssl.c |    4 +---
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/mailx.1 b/mailx.1
index 417ea04..a02e430 100644
--- a/mailx.1
+++ b/mailx.1
@@ -3575,7 +3575,7 @@ Only applicable if SSL/TLS support is built using OpenSSL.
 .TP
 .B ssl-method
 Selects a SSL/TLS protocol version;
-valid values are `ssl2', `ssl3', and `tls1'.
+valid values are `ssl3', and `tls1'.
 If unset, the method is selected automatically,
 if possible.
 .TP
diff --git a/openssl.c b/openssl.c
index b4e33fc..44fe4e5 100644
--- a/openssl.c
+++ b/openssl.c
@@ -216,9 +216,7 @@ ssl_select_method(const char *uhp)
 
 	cp = ssl_method_string(uhp);
 	if (cp != NULL) {
-		if (equal(cp, "ssl2"))
-			method = SSLv2_client_method();
-		else if (equal(cp, "ssl3"))
+		if (equal(cp, "ssl3"))
 			method = SSLv3_client_method();
 		else if (equal(cp, "tls1"))
 			method = TLSv1_client_method();
